src/nvim/ex_cmds.lua

```diff
@@ -2678,18 +2678,18 @@ return {
     func='ex_ni',
   },
   {
-    command='tearoff',
-    flags=bit.bor(NEEDARG, EXTRA, TRLBAR, NOTRLCOM, CMDWIN),
-    addr_type=ADDR_LINES,
-    func='ex_tearoff',
-  },
-  {
     command='terminal',
     flags=bit.bor(BANG, FILES, CMDWIN),
     addr_type=ADDR_LINES,
     func='ex_terminal',
   },
   {
+    command='tearoff',
+    flags=bit.bor(NEEDARG, EXTRA, TRLBAR, NOTRLCOM, CMDWIN),
+    addr_type=ADDR_LINES,
+    func='ex_tearoff',
+  },
```

Comment:
There is a list of special cases, if I recall, that overrides the alphabetical precedence here. But I'd rather just remove `:tearoff` and deal with this in 10 years when someone asks for the feature to be implemented. (We will not support `:tearoff` any time soon)